src/hbb/processors/categorizer.py

```diff
@@ -0,0 +1,329 @@
+
+        selection.add('2FJ', ak.num(fatjets, axis=1) == 2)
+        selection.add('not2FJ', ak.num(fatjets, axis=1) != 2)
+
+        candidatejet = fatjets[
+            (fatjets.pt > 200)
+            & (abs(fatjets.eta) < 2.5)
+            & fatjets.isTight 
+        ]
+
+        candidatejet = candidatejet[:, :2]
+        candidatejet = ak.firsts(candidatejet[ak.argmax(candidatejet.particleNet_XbbVsQCD, axis=1, keepdims=True)])
+
+        bvl = candidatejet.particleNet_XbbVsQCD
+
+        selection.add('minjetkin',
+            (candidatejet.pt >= 450)
+            & (candidatejet.pt < 1200)
+            & (candidatejet.msdcorr >= 40.)
+            & (candidatejet.msdcorr < 201.)
+            & (abs(candidatejet.eta) < 2.5)
+        )
+        selection.add('minjetkinmu',
+            (candidatejet.pt >= 400)
+            & (candidatejet.pt < 1200)
+            & (candidatejet.msdcorr >= 40.)
+            & (candidatejet.msdcorr < 201.)
+            & (abs(candidatejet.eta) < 2.5)
+        )
+        selection.add('jetid', candidatejet.isTight)
+        selection.add('ddbpass', (bvl >= 0.5))
+
+        jets = events.Jet
+        jets = jets[
+            (jets.pt > 30.)
+            & (abs(jets.eta) < 5.0)
+            & jets.isTight
+        ]
+
+        # only consider first 4 jets to be consistent with old framework
+        jets = jets[:, :4]
+        dphi = abs(jets.delta_phi(candidatejet))
+        selection.add('antiak4btagMediumOppHem', ak.max(jets[dphi > np.pi / 2].btagDeepFlavB, axis=1, mask_identity=False) < 0.3086) 
+        ak4_away = jets[dphi > 0.8]
+        selection.add('ak4btagMedium08', ak.max(ak4_away.btagDeepFlavB, axis=1, mask_identity=False) > 0.3086) 
+
+        met = events.MET
+        selection.add('met', met.pt < 140.)
+
+        # VBF specific variables                                                      
+        dR = jets.delta_r(candidatejet)
+        ak4_outside_ak8 = jets[dR > 0.8]
+
+        jet1 = ak4_outside_ak8[:, 0:1]
+        jet2 = ak4_outside_ak8[:, 1:2]
+
+        deta = abs(ak.firsts(jet1).eta - ak.firsts(jet2).eta)
+        mjj = ( ak.firsts(jet1) + ak.firsts(jet2) ).mass
+
+        isvbf = ((deta > 3.5) & (mjj > 1000))
+        isvbf = ak.fill_none(isvbf,False)
+        selection.add('isvbf', isvbf)
+
+        isnotvbf = ak.fill_none(~isvbf,True)
+        selection.add('notvbf', isnotvbf)
+
+        goodmuon = (
+            (events.Muon.pt > 10)
+            & (abs(events.Muon.eta) < 2.4)
+            & (events.Muon.pfRelIso04_all < 0.25)
+            & events.Muon.looseId
+        )
+        nmuons = ak.sum(goodmuon, axis=1)
+        leadingmuon = ak.firsts(events.Muon[goodmuon])
+
+        goodelectron = (
+            (events.Electron.pt > 10)
+            & (abs(events.Electron.eta) < 2.5)
+            & (events.Electron.cutBased >= 2)
+        )
+        nelectrons = ak.sum(goodelectron, axis=1)
+
+        ntaus = ak.sum(
```

Comment:
I vote that we should get rid of the number of taus cut, it causes more pain in the object review than benefit, we can always add it back if needed